sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/TableSchemaCache.java

```diff
@@ -271,10 +280,11 @@ public void refreshThread() {
       if (timeRemaining.getMillis() > 0) {
         Thread.sleep(timeRemaining.getMillis());
       }
-    } catch (InterruptedException e) {
-      runUnderMonitor(() -> this.stopped = true);
-      return;
-    } catch (IOException e) {
+    } catch (Exception e) {
```

Comment:
Is it appropriate to swallow _all_ exceptions? Could there be cases where the daemon needs to be monitored and restarted?

Reply:
That is sorta what is happening - we still exit the function, but schedule another closure. I'm not aware of any case where we want to shut down the executor itself.